src/pages/blog/experimental-server-side-rendering.md

```diff
@@ -0,0 +1,171 @@
+
+```html
+<script>
+  if(!parseCookie(document.cookie).auth) {
+    window.location = '/login';
+  }
+</script>
+```
+
+1. Call an API that will check if a user is logged in, like `/api/auth`. Going this route is going to mean that you need to show loading spinners in each of your islands while you wait on the response.
+
+```jsx
+function App() {
+  const auth = useAuth();
+  if(auth.loading) return <AnnoyingLoadingSpinner />
+  if(!auth.loggedIn) return <LoginPage />;
+  return <UserProfile user={auth.user} />
+}
+```
+
+No matter your approach, going client-side for authentication is always going to mean:
+
+- Things won’t run if JavaScript is disabled.
+- Your server can only respond with `200 OK` response codes.
+- It’s just plain slow. The user has to wait for the full HTML response (including JS, CSS, etc) even if they then immediately redirect. This creates multiple page requests before the user lands on the correct page.
+
+## Astro SSR: Designed to be Simple
+
+SSR might sound like a solved problem. Next.js, Nuxt, Gatsby, SvelteKit, and the whole gang of modern JavaScript meta-frameworks all have some idea of SSR already in place. SSR as a term is really just a rebranding of how Rails, PHP and other server frameworks already work today. If that’s all true, then what’s the big deal?
+
+Astro has one key advantage over the current set of JavaScript-focused meta-frameworks who have tackled SSR before us: **Astro was designed to run on the server.** 
+
+With Astro you are free to write your server code how it works in your mind, free from the unnecessary abstractions that come with other, “universal” SPA-first frameworks. While SPA-first thinking might be great for super stateful web applications (think dashboards, apps, portals) it’s a cost that most content-focused websites would be better off not paying. 
+
+Here’s how user authentication works in Astro SSR, using only 5 lines of JavaScript:
+
+```astro
+---
+// Example: src/pages/index.astro
+// In an Astro component (*.astro), you write server code directly 
+// in the component front matter (this space between the two --- fences). 
+
+// 1. Import any dependencies (Full support for JavaScript/TypeScript)
+import { getUser } from '../api/index.js';
+
+// 2. Check that the user token exists and is not malicious.
+const user = await getUser(Astro.request);
+
+// 3. If no user was found, return a redirect. This instantly
+//    completes the response with the correct status code & headers.
+if(!user) {
+  return Astro.redirect('/login');
+}
+
+// 4. If the user is logged in, you can now use the `user` object
+//    right in your page template to show an avatar, name, etc.
+---
+<html> <!-- contents here --> </html>
```

Comment:
```suggestion
<html>
...
<h1>Hello {user.name} 👋</h1>
<img src={user.avatar} alt={user.name}>
...
</html>
```

Nit: might be useful to show how simple Astro templating is. Some people with a nunjucks or jade background would appreciate it!